packages/core/src/shared/logger/toolkitLogger.ts

```diff
@@ -126,6 +108,31 @@ export class WinstonToolkitLogger implements Logger, vscode.Disposable {
               })
     }
 
+    public setTopic(topic: LogTopic = 'Unknown') {
+        this.topic = topic
+    }
+
+    /* Format the message with topic header */
+    private addTopicToMessage(message: string | Error): string | Error {
+        /*We shouldn't print unknow before current logging calls are migrated
+         * TODO: remove this once migration of current calls is completed
+         */
+        if (this.topic === 'Unknown') {
+            return message
+        }
+        const topicPrefix = `${this.topic}: `
+        if (typeof message === 'string') {
+            return topicPrefix + message
+        } else if (message instanceof Error) {
+            /* Create a new Error object to avoid modifying the original */
+            const topicError = new Error(topicPrefix + message.message)
```

Comment:
We have wrappers around errors that can have additional properties, see ToolkitError and the `code` property.

If possible can you modify the message on the given error, but I forget if that is read only.

Otherwise look to use ToolkitError.chain, and then make the `cause` the given error

Reply:
Thanks for pointing out the other extended errors. The messages appears to be read only, looking into the chain method.

Reply:
There are two problems with `ToolkitError.chain`:
1. it only applies to `toolkitErrors`, there are other custom error types that doesn't support chaining
2. the chaining is flaky because we have another `info` object defined within `toolkitErrors` with information such as `code` and `cause`. This causes duplication when printing out errors:
<img width="1066" alt="Screenshot 2024-09-18 at 11 10 49 AM" src="https://github.com/user-attachments/assets/ac55343c-5c04-495d-8980-135c035e6ed8">

Solution:
1. When adding header to errors, create another object to wrap around the error with topicHeader being the first entry. This cleanly appends the topic without any overhead, and can be applied to any type of error:
![screenshot_2024-09-18_at_10 43 19___am_720](https://github.com/user-attachments/assets/26b96758-2067-42f5-8f4f-8dda890980ea)
2. Remove the `info` object within the `toolkitErrors` constructor, keep the individual variables, so it doesn't get duplicated when printing.

Reply:
> it only applies to toolkitErrors, there are other custom error types that doesn't support chaining

I'm not sure I understand this, can't we put any error we want inside the chain as an argument?

But regardless I like the new `topic` field, its much cleaner. Is it possible to use `topicPrefix.split(':', 1)[0]` to trim the `: `?

Reply:
True, the chain would work on others as well, its just that we have to make the topic error a `toolkitError` as well.
It also has the problem of additional location statements (those `at` lines)

Glad you like the new field, will remove the `: `.